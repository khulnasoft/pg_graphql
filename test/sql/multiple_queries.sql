begin;

    create table account(
        id serial primary key,
        email varchar(255) not null,
        priority int
    );

    insert into account(email)
    values ('email_1'), ('email_2');

    -- Scenario: Two queries
    select jsonb_pretty(
        graphql.resolve($$
            {
              forward: accountCollection(orderBy: [{id: AscNullsFirst}]) {
                edges {
                  node {
                    id
                  }
                }
              }
              backward: accountCollection(orderBy: [{id: DescNullsFirst}]) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );

rollback
