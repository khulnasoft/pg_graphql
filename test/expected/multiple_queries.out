begin;
    create table account(
        id serial primary key,
        email varchar(255) not null,
        priority int
    );
    insert into account(email)
    values ('email_1'), ('email_2');
    -- Scenario: Two queries
    select jsonb_pretty(
        graphql.resolve($$
            {
              forward: accountCollection(orderBy: [{id: AscNullsFirst}]) {
                edges {
                  node {
                    id
                  }
                }
              }
              backward: accountCollection(orderBy: [{id: DescNullsFirst}]) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "forward": {           +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 1+
                     }          +
                 },             +
                 {              +
                     "node": {  +
                         "id": 2+
                     }          +
                 }              +
             ]                  +
         },                     +
         "backward": {          +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 2+
                     }          +
                 },             +
                 {              +
                     "node": {  +
                         "id": 1+
                     }          +
                 }              +
             ]                  +
         }                      +
     }                          +
 }
(1 row)

rollback
